### PR TITLE
Clean-up of unused variables, other es6 linter complaints

### PR DESCRIPTION
**Pull request by jasonpropp**

### CODE REVIEW

File: src/listener/utils/resource-transforms.js [hunk 1]
```diff
@@ -49,6 +49,10 @@ function addPeriod(text) {
   return text;
 }
 
+function isVowel(x) {
```
Comment (danielreed) [inline src/listener/utils/resource-transforms.js]: where is this used?

File: src/listener/utils/transforms/resource-transformer.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ THE SOFTWARE.
 */
 
 import ServerProfile from './server-profile';
+import ServerProfileTemplate from './server-profile-template';
```
Comment (danielreed) [inline src/listener/utils/transforms/resource-transformer.js]: good catch on this.

File: src/middleware/utils/lexer.js [hunk 3]
```diff
@@ -22,14 +22,6 @@ THE SOFTWARE.
 
 import FuzzySet from 'fuzzyset.js';
 
-const defaults = {
```
Comment (danielreed) [inline src/middleware/utils/lexer.js]: these aren't used anywhere in the lexer?

File: src/listener/utils/resource-transforms.js (hunk 1 above)
Comment (jasonpropp) [inline src/listener/utils/resource-transforms.js]: line 162 of this file, it was being called but wasn't actually a method anywhere

**Pull request merged**